dashboard/app/models/user.rb

```diff
@@ -615,7 +615,7 @@ def email_required?
   end
 
   def username_required?
-    provider == User::PROVIDER_MANUAL
+    provider == User::PROVIDER_MANUAL || username_changed?
```

Comment:
I agree, that's a little strange, and I also wasn't sure if there was any good reason not to just always require a username (are there even registration paths that would let that happen anymore, or is it just legacy users?). Let's discuss this at some point - I wanted to hold off on any changes here until discussing the implications with you.